src/Plugin/Commerce/PaymentGateway/Heartland.php

```diff
@@ -0,0 +1,391 @@
+        try {
+            if ($capture) {
+                $response = $card->charge($number)
+                    ->withCurrency($currency)
+                    ->execute();
+            } else {
+                $response = $card->authorize($number)
+                    ->withCurrency($currency)
+                    ->execute();
+            }
+
+            $remote_id = $response->transactionId;           
+            $next_state = $capture ? 'completed' : 'authorization';    
+            
+            // Commerce plugin sets state and remote Id for next step
+            $payment->setState($next_state);
+            $payment->setRemoteId($remote_id);
+            $payment->save();
+        } catch (GatewayException $e) {
+            throw new PaymentGatewayException($e->getMessage());
+        }
+    }
+  
+    /**
+     * {@inheritdoc}
+     */
+    public function capturePayment(PaymentInterface $payment, Price $amount = null)
+    {
+        $this->assertPaymentState($payment, ['authorization']);
+        // If not specified, capture the entire amount.
+        $amount = $amount ?: $payment->getAmount();
+  
+        // Perform the capture request here, throw an exception if it fails.
+        // See \Drupal\commerce_payment\Exception for the available exceptions.
+
+        // Set some variables to use for our transaction
+        $remote_id = $payment->getRemoteId();
+        $number = $amount->getNumber();
+        $currency = $amount->getCurrencyCode();
+
+        // Heartland SDK - Capture Authorization
+        // https://developer.heartlandpaymentsystems.com/Documentation/credit-card-payments/#capture-an-authorization
+        try {
+            $response = Transaction::fromId($remote_id)
+                ->capture($number)
+                ->withCurrency($currency)
+                ->execute();
+
+            // Commerce plugin sets state and amount for next step
+            $payment->setState('completed');
+            $payment->setAmount($amount);
+            $payment->save();
+        } catch (GatewayException $e) {
+            throw new PaymentGatewayException($e->getMessage());
+        }
+    }
+  
+    /**
+     * {@inheritdoc}
+     */
+    public function voidPayment(PaymentInterface $payment)
+    {
+        // The voidPayment function is called when 'Void' is clicked on the orders payment screen,
+        // not 'Delete' like the Drupal Commerce docs state
+        $this->assertPaymentState($payment, ['authorization']);
+      
+        // Perform the void request here, throw an exception if it fails.
+        // See \Drupal\commerce_payment\Exception for the available exceptions.
+
+        // Set some variables to use for our transaction
+        $remote_id = $payment->getRemoteId();
+        $amount = $payment->getAmount();
+        $number = $amount->getNumber();
+        $currency = $amount->getCurrencyCode();
```

Comment:
Let's remove the `$amount`, `$number`, and `$currency` variables if they're not needed here.